src/lib/plugin_apis/nvme.api

```diff
@@ -0,0 +1,647 @@
+ * @BD_NVME_TRANSPORT_TYPE_TCP: TCP Transport
+ * @BD_NVME_TRANSPORT_TYPE_LOOP: Intra-host Transport (loopback)
+ */
+/* BpG-skip-end */
+typedef enum {
+    BD_NVME_TRANSPORT_TYPE_UNSPECIFIED = 0,
+    BD_NVME_TRANSPORT_TYPE_RDMA        = 1,
+    BD_NVME_TRANSPORT_TYPE_FC          = 2,
+    BD_NVME_TRANSPORT_TYPE_TCP         = 3,
+    BD_NVME_TRANSPORT_TYPE_LOOP        = 254
+} BDNVMETransportType;
+
+#define BD_NVME_TYPE_ERROR_LOG_ENTRY (bd_nvme_error_log_entry_get_type ())
+GType bd_nvme_error_log_entry_get_type ();
+
+/**
+ * BDNVMEErrorLogEntry:
+ * @error_count: internal error counter, a unique identifier for the error.
+ * @command_id: the Command Identifier of the command that the error is associated with or %0xffff if the error is not specific to a particular command.
+ * @command_specific: Command Specific Information specific to @command_id.
+ * @command_status: the Status code for the command that completed.
+ * @command_error: translated command error in the BD_NVME_ERROR domain or %NULL in case @command_status indicates success.
+ * @lba: the first LBA that experienced the error condition.
+ * @nsid: the NSID of the namespace that the error is associated with.
+ * @transport_type: type of the transport associated with the error.
+ */
+typedef struct BDNVMEErrorLogEntry {
+    guint64 error_count;
+    guint16 command_id;
+    guint64 command_specific;
+    guint16 command_status;
+    GError *command_error;
+    guint64 lba;
+    guint32 nsid;
+    BDNVMETransportType transport_type;
+} BDNVMEErrorLogEntry;
+
+/**
+ * bd_nvme_error_log_entry_free: (skip)
+ * @entry: (allow-none): %BDNVMEErrorLogEntry to free
+ *
+ * Frees @entry.
+ */
+void bd_nvme_error_log_entry_free (BDNVMEErrorLogEntry *entry) {
+    if (entry == NULL)
+        return;
+
+    if (entry->command_error)
+        g_error_free (entry->command_error);
+    g_free (entry);
+}
+
+/**
+ * bd_nvme_error_log_entry_copy: (skip)
+ * @entry: (allow-none): %BDNVMEErrorLogEntry to copy
+ *
+ * Creates a new copy of @entry.
+ */
+BDNVMEErrorLogEntry * bd_nvme_error_log_entry_copy (BDNVMEErrorLogEntry *entry) {
+    BDNVMEErrorLogEntry *new_entry;
+
+    if (entry == NULL)
+        return NULL;
+
+    new_entry = g_new0 (BDNVMEErrorLogEntry, 1);
+    memcpy (new_entry, entry, sizeof (BDNVMEErrorLogEntry));
+    if (entry->command_error)
+        new_entry->command_error = g_error_copy (entry->command_error);
+
+    return new_entry;
+}
+
+GType bd_nvme_error_log_entry_get_type () {
+    static GType type = 0;
+
+    if (G_UNLIKELY (type == 0)) {
+        type = g_boxed_type_register_static ("BDNVMEErrorLogEntry",
+                                             (GBoxedCopyFunc) bd_nvme_error_log_entry_copy,
+                                             (GBoxedFreeFunc) bd_nvme_error_log_entry_free);
+    }
+    return type;
+}
+
+
+/**
+ * bd_nvme_get_controller_info:
+ * @device: a NVMe controller device (e.g. /dev/nvme0)
+ * @error: (out): place to store error (if any)
```

Comment:
You should mark the `error` optional (and make it optional if it isn't), see https://github.com/storaged-project/libblockdev/issues/527 (applies to all functions).

Reply:
Changed.